src/hotspot/share/code/dependencies.cpp

```diff
@@ -1332,106 +1329,50 @@ Klass* ClassHierarchyWalker::find_witness_anywhere(Klass* context_type,
   bool do_counts = count_find_witness_calls();
```

Comment:
Does count_find_witness_calls() have side effects and required to be called here in product? If not,
`do_counts` is used only for not_product code, so consider enclose it into  NOT_PRODUCT() here and where it is used.

Reply:
`count_find_witness_calls()` is a no-op in product binaries:
```
#ifndef PRODUCT
...
static bool count_find_witness_calls() {
...
#else
#define count_find_witness_calls() (0)
#endif //PRODUCT
```